pkg/v2/crud/traverse.go

```diff
@@ -79,10 +81,32 @@ func (t traverser) traverseSelectedElements(query *expr.Expression) error {
 
 func (t traverser) traverseQualifiedElements(filter *expr.Expression) error {
 	return t.nav.ForEachChild(func(index int, child prop.Property) error {
-		t.nav.At(index)
-		if err := t.nav.Error(); err != nil {
-			return err
+		// index is not trustworthy if an earlier child has been deleted by
```

Comment:
This change is necessary to allow dealing with deletion of an array element as the array elements are being visited to see if they pass the filter.